src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -127,6 +127,22 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     }
   }
 
+  def addUserToPolicy(resourceType: ResourceType, policyName: AccessPolicyName, resource: Resource, email: String, userInfo: UserInfo) = {
```

Comment:
this function (and remove) should take a ResourceAndPolicyName as a parameter.

Also you should be explicit about return types of public functions. And when you do you will find that this is not returning what think it is.